### PR TITLE
fixed vertical scrolling on mobile issue with 3 steps.

### PR DESCRIPTION
**Pull request by chriswnewman**
This PR fixes vertical scrolling issues on mobile.

1) Added a DI provider with an instance of HammerGestureConfig per https://github.com/angular/angular/issues/10541#issuecomment-300761387

2) App Modules need to import CarouselModule using static forRoot() method. **This may be considered a "breaking change"**
- https://alligator.io/angular/providers-shared-modules/ 

3) Ignore vertical pans in the onPan function, where y velocity is greater than x velocity.
 - https://github.com/angular/angular/issues/10541#issuecomment-346539242

### CODE REVIEW

File: projects/carousel/src/lib/carousel.component.ts [hunk 1]
```diff
@@ -247,6 +247,12 @@ export class MatCarouselComponent
   }
 
   public onPan(event: any, slideElem: HTMLElement): void {
+    // https://github.com/angular/angular/issues/10541#issuecomment-346539242
+    // if y velocity is greater, it's a panup/pandown, so ignore.
+    if (Math.abs(event.velocityY) > Math.abs(event.velocityX)) {
+      // console.log('y velocity is greater, ignoring');
```
Comment (gbrlsnchs) [inline projects/carousel/src/lib/carousel.component.ts]: Could you remove this line?

File: projects/demo/src/app/app.module.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ import { AppComponent } from './app.component';
     MatButtonModule,
     MatCardModule,
     MatCheckboxModule,
-    MatCarouselModule,
+    MatCarouselModule.forRoot(),
```
Comment (gbrlsnchs) [inline projects/demo/src/app/app.module.ts:29]: If `forRoot` is the way to import this module now, maybe it should be a nice idea to explain that in README.

**Review by gbrlsnchs — CHANGES_REQUESTED** [review]

File: projects/carousel/src/lib/carousel.component.ts (hunk 1 above)
Comment (chriswnewman) [reply projects/carousel/src/lib/carousel.component.ts]: Removed line.

File: projects/demo/src/app/app.module.ts (hunk 2 above)
Comment (chriswnewman) [reply projects/demo/src/app/app.module.ts:29]: Updated README.md

**Pull request merged**